src/System.Private.ServiceModel/src/System/ServiceModel/Channels/StreamedFramingRequestChannel.cs

```diff
@@ -39,14 +39,19 @@ private byte[] Preamble
             get { return _startBytes; }
         }
 
+        protected internal override Task OnOpenAsync(TimeSpan timeout)
+        {
+            return TaskHelpers.CompletedTask();
+        }
+
         protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return new CompletedAsyncResult(callback, state);
+            throw ExceptionHelper.PlatformNotSupported("StreamedFramingRequestChannel async path");
```

Comment:
What happens if a derived class implements OnBeginOpen and calls base.OnBeginOpen()?